R/plotAbundance.R

```diff
@@ -164,36 +160,36 @@ setMethod("plotAbundance", signature = c("SummarizedExperiment"),
             stop("No data to plot. nrow(x) == 0L.", call. = FALSE)
         }
         .check_assay_present(assay.type, x)
-        # if rank is set to NULL, default to plotExpression 
-        if(is.null(rank)){
-            plot <- plotExpression(x, features = features, 
-                                exprs_values = assay.type,
-                                one_facet = one_facet,
-                                ncol = ncol, scales = scales, ...)
-            ylab <- gsub("Expression","Abundance",plot$labels$y)
-            plot <- plot +
-                ylab(ylab)
-            return(plot)
-        }
         ############################# INPUT CHECK #############################
-        if(!.is_non_empty_string(rank)){
-            stop("'rank' must be an non empty single character value.",
+        if(!.is_non_empty_string(rank) && !is.null(rank)){
+            stop("'rank' must be an non empty single character value or NULL.",
                 call. = FALSE)
         }
         if(!.is_a_bool(use_relative)){
             stop("'use_relative' must be TRUE or FALSE.",
                 call. = FALSE)
         }
-        .check_taxonomic_rank(rank, x)
+        if(!is.null(rank)){
+            .check_taxonomic_rank(rank, x)
+        }
         .check_for_taxonomic_data_order(x)
         layout <- match.arg(layout, c("bar","point"))
         order_rank_by <- match.arg(order_rank_by, c("name","abund","revabund"))
         .check_abund_plot_args(one_facet = one_facet,
                             ncol = ncol)
-        if( !is.null(features) ){
+        if( !is.null(features)){
             features <- match.arg(features, colnames(colData(x)))
         }
         ########################### INPUT CHECK END ###########################
+        if(is.null(rank)){
+           rank <- "Feature"
+           rowData(x)[[ rank ]] <- rownames(x)
+        }
+        # If more than 500 features, revert to first taxonomic rank
+        if(is.null(rank) && nrow(assay(x,assay.type=assay_name))>500){
+            rank = taxonomyRanks(x)[1]
+        }
+        
```

Comment:
Problem with this approach is that instead of doing what user wants, this does something else. Moreover, some data might not have taxonomy ranks.

That is why I suggest that after line 193, you check how many rows there are in the abundance data. That is because there might be many features even though rank is specified.

If there is over 100 features (should be enough), give error. "The data contains more than 100 rows. Abundance plot cannot be created." or something like that